docs/HARDWARE-BRINGUP.md

```diff
@@ -50,6 +50,108 @@ One word of caution: a lot of ARM board (like the ever popular Raspberry Pi) act
 
 There are two subtle points when it comes to getting reliable console with QEMU/ARM64 emulation. First of all, QEMU only emulates pl011 UART (which means that things like earlyprintk on UART8250 as was described above won't work). Second point is that it is rather [futile](https://unix.stackexchange.com/questions/479085/can-qemu-m-virt-on-arm-aarch64-have-multiple-serial-ttys-like-such-as-pl011-t) to look for more than one pl011 UART with QEMU.
 
+## 3. Figuring out your firmware/bootloader situation
+
+Please refer to the [BIOS/Firmware management doc](BIOS-FIRMWARE.md) if you are interested in the generic introduction. A few useful tricks that may come handy are:
+
+* use custom/vendor provided device trees - just drop them into config partition as `eve.dtb`
+* `grub.cfg` in the config partition is very helpful to deal with one-off issues to provide a rapid edit-compile-run cycle
+* initrd from the installer phase can be used for as long as you don't have Linux kernel with storage support
+
+## 4. Bringing up basic Linux kernel
+
+It is recommended to focus on bringing up the basic Linux kernel first before you can proceed with the rest of hardware enablement. Doing so will allow you to use it as a tool to inspect your hardware further even in situation where you may not have working drivers for any kind of storage (see the section above on using initrd). The usual sequence is:
+
+* Enabling at least one kind of persistent storage medium
+* Enabling KVM and Xen hypervisors
+* Enabling the rest of the hardware
+
+If the last point sounds a lot like [draw the rest of the owl](https://knowyourmeme.com/memes/how-to-draw-an-owl) that's because it largely is. The good news is that unlike most traditional Linux distributions, EVE offloads a lot of the hardware management to user VMs running on top of it. All that EVE needs to do is make sure it can virtualize the buses that hardware is connected to AND enable a few critical hardware pieces for its own use:
+
+* Wired networking (this allows EVE to talk to its controller and update its rootfs images via the network)
+* TPM (or any other hardware trust element)
+* Watchdog
```

Comment:
If you have some wisdom to impart on the topic of watchdog timers it would be great to have a sub-section on that in the document. 

Maybe structure that in a way so we can later add info about how to make EVE use a partilar LED, thus have some section after section 5?

Reply:
Yup -- that's the plan @eriknordmark -- to keep adding to these sections like that based on experience.